assets/src/scripts/monitoring-location/components/hydrograph/index.js

```diff
@@ -96,12 +97,17 @@ export const attachToNode = function(store,
         store.dispatch(setSelectedIVMethodID(timeSeriesId));
     }
 
-    // Fetch waterwatch flood levels - TODO: consider only fetching when gage height is requested
-    store.dispatch(floodDataActions.retrieveWaterwatchData(siteno));
+    // Fetch waterwatch flood levels
+    const fetchFloodLevelsPromise = store.dispatch(floodDataActions.retrieveWaterwatchData(siteno));
 
-    fetchDataPromise.then(() => {
+    let fetchDataPromises = [fetchHydrographDataPromise];
+    // If flood levels are to be shown then wait to render the hydrograph until those have been fetched.
+    if (parameterCode === config.GAGE_HEIGHT_PARAMETER_CODE) {
+        fetchDataPromises.push(fetchFloodLevelsPromise);
+    }
+    Promise.all(fetchDataPromises).then(() => {
+        console.log('Done fetching data');
```

Comment:
Seems like this log statement might be intentional, but not sure.

Reply:
Removed